ntoskrnl/io/iomgr/deviface.c

```diff
@@ -304,7 +305,177 @@ IoGetDeviceInterfaceAlias(IN PUNICODE_STRING SymbolicLinkName,
+    {
+        DPRINT1("RtlAppendUnicodeStringToString() failed with status 0x%08lx\n", Status);
+        goto cleanup;
+    }
+
+    Status = IoOpenDeviceInterfaceRegistryKey(&BaseKeyName,
+                                              GENERIC_READ,
+                                              &InterfaceKey);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("IoOpenDeviceInterfaceRegistryKey() failed with status 0x%08lx\n", Status);
+        goto cleanup;
+    }
+
+    Status = ZwQueryKey(InterfaceKey,
+                        KeyFullInformation,
+                        NULL,
+                        0,
+                        &Size);
+    if (!NT_SUCCESS(Status) && Status != STATUS_BUFFER_TOO_SMALL)
+    {
+        DPRINT1("ZwQueryKey() failed with status 0x%08lx\n", Status);
+        goto cleanup;
+    }
+
+    fip = ExAllocatePool(PagedPool, Size);
+    if (!fip)
+    {
+        DPRINT1("ExAllocatePool() failed\n");
+        goto cleanup;
+    }
+
+    Status = ZwQueryKey(InterfaceKey,
+                        KeyFullInformation,
+                        fip,
+                        Size,
+                        &Size);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("ZwQueryKey() failed with status 0x%08lx\n", Status);
+        goto cleanup;
+    }
+
+    /* Enumerate subkeys (i.e. the different device objects) */
+    for (i = 0; i < fip->SubKeys; i++)
+    {
+        Status = ZwEnumerateKey(InterfaceKey,
+                                i,
+                                KeyBasicInformation,
+                                NULL,
+                                0,
+                                &Size);
+        if (Status == STATUS_NO_MORE_ENTRIES)
+        {
+            break;
+        }
+        else if (!NT_SUCCESS(Status) && Status != STATUS_BUFFER_TOO_SMALL)
+        {
+            DPRINT1("ZwEnumerateKey() failed with status 0x%08lx\n", Status);
+            goto cleanup;
+        }
+
+        bip = ExAllocatePool(PagedPool, Size);
+        if (!bip)
+        {
+            DPRINT1("ExAllocatePool() failed\n");
+            goto cleanup;
+        }
+
+        Status = ZwEnumerateKey(InterfaceKey,
+                                i,
+                                KeyBasicInformation,
+                                bip,
+                                Size,
+                                &Size);
+        if (!NT_SUCCESS(Status))
+        {
+            DPRINT1("ZwEnumerateKey() failed with status 0x%08lx\n", Status);
+            goto cleanup;
+        }
+
+        /* Check bip->Name */
+        RtlInitUnicodeString(&bipName, NULL);
+        RtlInitUnicodeString(&InLinkName, NULL);
+
+        bipName.Length = wcslen(bip->Name);
+        bipName.MaximumLength = wcslen(bip->Name);
+        bipName.Buffer = bip->Name;
+
+        AliasKeyName.Length = 0;
+        AliasKeyName.MaximumLength += bipName.Length;
+        AliasKeyName.Buffer = ExAllocatePool(PagedPool, AliasKeyName.MaximumLength);
```

Comment:
The fact that a kernel mode driver fails to free `AliasSymbolicLinkName` is most likely because you're supplying to that variable with data in memory allocated that is pageable, that is, it's not guaranteed to stay in the physical memory, unlike with non-paged pools.

To further confirm the case, the Windows' implementation of this function allocates data in memory as non-paged.

Reply:
OK, I tried with `NonPagedPool`, but it still doesn't help. :frowning_face: The same exception still occurs during loading the drivers, and the backtrace is also nearly the same.
Now the only difference is `DbgUserBreakPoint` from ntoskrnl in `Eip`:
`<ntoskrnl.exe:18234e (:0 (DbgUserBreakPoint))>`